lib/uiautomator2.js

```diff
@@ -97,6 +97,32 @@ class UiAutomator2Server {
     return resultInfo;
   }
 
+  /**
+   * @typedef {Object} PackageInfo
+   * @property {string} installState
+   * @property {string} appPath
+   * @property {string} appId
+   */
+  /**
+   * @param {PackageInfo[]} packagesInfo
+   * @returns {boolean}
+   */
+  shouldUninstallServerPackages(packagesInfo = []) {
+    return (packagesInfo.some(({installState}) => installState === this.adb.APP_INSTALL_STATE.NEWER_VERSION_INSTALLED)
```

Comment:
I am not sure about this change.

This is the situation we were checking before with this condition:

- if one of server components is installed and the other is not then we want to uninstall the existing one before installing both

If any of the components has a newer version installed then we don't need to remove it unless the current uia2 driver is not the newest one

Reply:
I have read the PR description and confirmed the case addresses my last guess above.
In such case and also in order to preserve the legacy behavior the uninstall check should be changed to

```js
return (packagesInfo.some(({installState}) => installState !== this.adb.APP_INSTALL_STATE.NOT_INSTALLED)
    && !packagesInfo.every(({installState}) => installState === this.adb.APP_INSTALL_STATE.NOT_INSTALLED));
```

Reply:
also the shouldUninstallServerPackages  method itself should have a proper doctrings explaining its purpose since this condition is not very readable